src/main/java/org/testng/DependencyMap.java

```diff
@@ -60,16 +67,16 @@ public ITestNGMethod getMethodDependingOn(String methodName, ITestNGMethod fromM
     if (l.isEmpty() && fromMethod.ignoreMissingDependencies()) {
       return fromMethod;
     }
-    for (ITestNGMethod m : l) {
-      // If they are in the same class hierarchy, they must belong to the same instance,
-      // otherwise, it's a method depending on a method in a different class so we
-      // don't bother checking the instance.
-      if (isSameInstance(fromMethod, m)
-          || belongToDifferentClassHierarchy(fromMethod, m)
-          || hasInstance(fromMethod, m)) {
-        return m;
-      }
+    Optional<ITestNGMethod> found = l.stream()
+        .parallel()
+        .filter(m -> isSameInstance(fromMethod, m)
+            || belongToDifferentClassHierarchy(fromMethod, m)
+            || hasInstance(fromMethod, m))
+        .findFirst();
+    if (found.isPresent()) {
```

Comment:
You can use `orElseThrow`

Reply:
I tried that, and for some reason it wasnt working. Thats why i fell back to this approach.